src/components/landing-layout/navigation/sign-in-button.tsx

```diff
@@ -15,13 +16,27 @@ import { ShowIf } from '../../common/show-if/show-if'
 
 export type SignInButtonProps = Omit<ButtonProps, 'href'>
 
+const INTERACTIVE_LOGIN_METHODS = ['internal', 'ldap', 'openid']
+
 export const SignInButton: React.FC<SignInButtonProps> = ({ variant, ...props }) => {
   const { t } = useTranslation()
-  const anyAuthProviderActive = useSelector((state: ApplicationState) => Object.values(state.config.authProviders)
-                                                                               .includes(true))
+  const authProviders = useSelector((state: ApplicationState) => state.config.authProviders, equal)
+  const [loginLink, setLoginLink] = useState('/login')
+
+  useEffect(() => {
+    const activeProviders = Object.entries(authProviders)
+                                  .filter(entry => entry[1] === true)
+                                  .map(entry => entry[0])
+    const activeOneClickProviders = activeProviders.filter(entry => !INTERACTIVE_LOGIN_METHODS.includes(entry))
+
+    if (activeProviders.length === 1 && activeOneClickProviders.length === 1) {
+      setLoginLink(`/api/v2/auth/${activeOneClickProviders[0]}`)
```

Comment:
the /api/v2/ should be imported. I think it's configured in the api classes.

Reply:
Done